### ISSUE TITLE
`folder_contents` breaks if a catalog column name matches any blob fields

### ISSUE THREAD

**Issue by sauzher** [plone/Products.CMFPlone#4078]
## BUG/PROBLEM REPORT 

`folder_contents` view breaks if a there is  a catalog column (metadata) that matches any blob fields name and that folder contains items with that blob field populated.

### What I did:

Add a portal_catalog metadata column named as any blob field (for example `image`)
Add an Image content in a folder (populate it with an actual file)
Try to access to folder_contents view

### What I expect to happen:
See folder contents items

### What actually happened:
The view stucks in loading mode. 

The backend raises an error:
```
2024-12-11 04:28:21 ERROR Zope.SiteErrorLog 1733909301.730.405219910046 http://localhost:8081/Plone/folder/@@getVocabulary
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module plone.app.content.browser.vocabulary, line 265, in __call__
  Module plone.app.content.utils, line 24, in json_dumps
  Module simplejson, line 412, in dumps
  Module simplejson.encoder, line 296, in encode
  Module simplejson.encoder, line 378, in iterencode
ValueError: Circular reference detected
```

Attributes list required by the pattern are filled with all catalog metadata columns

Python repr of the an item is something like 
```
{'CreationDate': '2024-12-06T11:12:43-05:00',
[ ... ]
 'getMimeIcon': '/Plone/++resource++mimetype.icons/txt.png',
 'getObjSize': '890.0 KB',
 'getURL': 'http://localhost:8081/Plone/esibizione-1',
'id': 'esibizione-1',

 'image': <plone.namedfile.file.NamedBlobImage object at 0x7f5babcf17d0>,

[ ... ]

 'total_comments': 0}
```
`image` value is not json serializable

### What version of Plone I am using:

Plone  5.x, 6.x Classic

**Comment by davisagli** [comment]
It's unusual to store a NamedBlobImage in catalog metadata. I wouldn't be surprised if it causes problems in other places too. Do you need to do it?

**Comment by sauzher** [comment]
@davisagli I totally agree. I found this situation during a migration from plone43 Archetypes to plone5 Dexterity.
I think that column was only used as column in atct-topic configuration, and it's not really needed anymore.

I was wondering whether there should be some sort of "do not add metadata names that references blob fields" disclaimer in the zcatalog (generally Plone) documentation.

**Comment by stevepiercy** [comment]
> I was wondering whether there should be some sort of "do not add metadata names that references blob fields" disclaimer in the zcatalog (generally Plone) documentation.

There is a relevant open issue https://github.com/plone/documentation/issues/1730 that needs many subject matter experts to take just one tiny, wafer-thin, after-dinner mint bite of it. Currently there is nothing useful in about zcatalog [Plone 6 Documentation](https://6.docs.plone.org/search.html?q=zcatalog). 😢

**Comment by rohnsha0** [comment]
> I was wondering whether there should be some sort of "do not add metadata names that references blob fields" disclaimer in the zcatalog (generally Plone) documentation.

In addition to this, I will suggest adding a check in the form to include a `warning` and `disallow` adding if catalog column name  matches any blob field

**Comment by rohnsha0** [comment]
Is this still valid?

**Comment by jensens** [comment]
Verified still present. custom_json_handler in plone/app/content/utils.py has no handling for NamedBlobImage/NamedBlobFile. Blob catalog metadata columns cause JSON serialization errors.
